x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/action/token/TransportCreateTokenActionTests.java

```diff
@@ -267,4 +269,25 @@ public void testKerberosGrantTypeCreatesWithRefreshToken() throws Exception {
             assertNotNull(sourceMap.get("refresh_token"));
         }
     }
+
+    public void testKerberosGrantTypeWillFailOnBase64DecodeError() throws Exception {
+        final TokenService tokenService = new TokenService(SETTINGS, Clock.systemUTC(), client, license, securityContext,
+            securityIndex, securityIndex, clusterService);
+        Authentication authentication = new Authentication(new User("joe"), new Authentication.RealmRef("realm", "type", "node"), null);
+        authentication.writeToContext(threadPool.getThreadContext());
+
+        final TransportCreateTokenAction action = new TransportCreateTokenAction(threadPool,
+            mock(TransportService.class), new ActionFilters(Collections.emptySet()), tokenService,
+            authenticationService, securityContext);
+        final CreateTokenRequest createTokenRequest = new CreateTokenRequest();
+        createTokenRequest.setGrantType("_kerberos");
+        createTokenRequest.setKerberosTicket(new SecureString("(:I".toCharArray()));
+
+        PlainActionFuture<CreateTokenResponse> tokenResponseFuture = new PlainActionFuture<>();
```

Comment:
We could wrap this in an `[assertListenerIsOnlyCalledOnce](https://github.com/elastic/elasticsearch/blob/51d0bcacdf0c2daf47ae1f17574e1884304a8271/x-pack/plugin/identity-provider/src/internalClusterTest/java/org/elasticsearch/xpack/idp/saml/sp/SamlServiceProviderIndexTests.java#L243)

Reply:
Thanks. This is a nice addition to the assertions.